File: witcher-system-presentation/src/main/java/com/nevdev/witcher/models/TaskViewModel.java
package com.nevdev.witcher.models;

import com.nevdev.witcher.core.Task;
import com.nevdev.witcher.core.User;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class TaskViewModel extends Task {
    private User customer;
    private User witcher;

    public TaskViewModel(Task task, User customer, User witcher) {
        super(task);
        this.customer = customer;
        this.witcher = witcher;
    }
}

File: witcher-system-ui/src/app/models/task/task.view.model.ts
import {TaskLocationModel} from './task.location.model';
import {TaskUserModel} from './task.user.model';
import {TaskBeastModel} from './task.beast.model';
import {TaskRewardModel} from './task.reward.model';

export class TaskViewModel {
  id: number;
  title: string;
  locationComment: string;
  location: TaskLocationModel;
  createOn: string;
  completionOn: string;
  done: boolean;
  paid: boolean;
  reward: TaskRewardModel;
  beasts: TaskBeastModel[];
  witchers: TaskUserModel[];
  customer: TaskUserModel;
  witcher: TaskUserModel;

  checkedLocation: string;
  checkedCurrency: string;
  checkedRewardValue: string;
  checkedBeast: string;
}
